### PR TITLE
Implement SDL handler

### PR DESCRIPTION
**Pull request by Megamouse**
Known issues:
- [x] ~~Config only cares for pad name, which might not be unique~~ won't implement for now
- [x] Add proper description
- [x] Fix rumble nonsense
- [x] Fix xbox controller nonsense
- [x] Rumble doesn't work properly
- [x] Crashes sometimes when SDL is initialized or closed
- [x] Fix HAVE_SDL2

### CODE REVIEW

File: rpcs3/Emu/Io/Null/NullPadHandler.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class NullPadHandler final : public PadHandlerBase
 		return true;
 	}
 
-	void ThreadProc() override
+	void process() override
```
Comment (kd-11) [inline rpcs3/Emu/Io/Null/NullPadHandler.h:67]: Maybe call it "run" or operator ()
Usually proc in the context of an executor (such as threadproc) means procedure and not process.

Comment (Megamouse) [reply rpcs3/Emu/Io/Null/NullPadHandler.h:67]: It's not a thread though. It's basically just one snapshot of the current state of each handler

Comment (kd-11) [reply rpcs3/Emu/Io/Null/NullPadHandler.h:67]: :ok_hand:

Comment (kd-11) [reply rpcs3/Emu/Io/Null/NullPadHandler.h:67]: thing->process sounds weird, but its just a preference at the end of the day.

File: rpcs3/Emu/System.cpp [hunk 2]
```diff
@@ -2291,7 +2291,7 @@ std::shared_ptr<utils::serial> Emulator::Kill(bool allow_autoexit, bool savestat
 		for (uint i = 0; thread_ctrl::state() != thread_state::aborting;)
 		{
 			// We don't need accurate timekeeping, using clocks may interfere with debugging
-			if (i >= (savestate ? 2000 : 1000))
+			if (i >= (savestate ? 2000u : 1000u))
```
Comment (elad335) [inline rpcs3/Emu/System.cpp:2294]: or simply change the type of i to int.

**Comment by oresttokovenko** [comment]
Can someone merge this?

**Comment by AniLeo** [comment]
Of course it's not going to be merged before it's completed. The author has merge permissions so they can do it themselves when it's ready.

**Comment by oresttokovenko** [comment]
> Of course it's not going to be merged before it's completed. The author has merge permissions so they can do it themselves when it's ready.

Sorry! I thought it was completed.

**Comment by Megamouse** [comment]
Merging without linux support for now since there seem to be linker issues with system SDL

**Pull request merged**